api/v1beta1/common_types.go

```diff
@@ -112,120 +112,32 @@ type BarbicanComponentTemplate struct {
-	// HMAC Key Type
-	HMACKeyType string `json:"HMACKeyType"`
+	// SecretStorePKCS11 -
+	SecretStorePKCS11 SecretStore = "pkcs11"
 
-        // +kubebuilder:validation:Optional
-	// +kubebuilder:default=CKM_GENERIC_SECRET_KEY_GEN
-	// HMAC Keygen Mechanism
-	HMACKeygenMechanism string `json:"HMACKeygenMechanism"`
-
-        // +kubebuilder:validation:Optional
-	// +kubebuilder:default=CKM_SHA256_HMAC
-	// HMAC Mechanism. This replaces hsm_keywrap_mechanism
-	HMACMechanism string `json:"HMACMechanism"`
-
-        // +kubebuilder:validation:Optional
-        // +kubebuilder:default=4
-        // +kubebuilder:validation:Maximum=7
-        // +kubebuilder:validation:Minimum=0
-        // Level of logging, where 0 means "no logging" and 7 means "debug".
-        LoggingLevel int `json:"loggingLevel"`
-
-	// +kubebuilder:validation:Required
-	// The HSM's IPv4 address (X.Y.Z.K)
-	ServerAddress string `json:"serverAddress"`
-
-	// +kubebuilder:validation:Optional
-	// The IP address of the client connecting to the HSM (X.Y.Z.K)
-	ClientAddress string `json:"clientAddress,omitempty"`
+	// DefaultP11ClientDataPath
+	DefaultP11ClientDataPath = "/etc/hsm-client"
+)
 
+// BarbicanPKCS11Template - Includes common HSM properties
+type BarbicanPKCS11Template struct {
         // +kubebuilder:validation:Required
         // OpenShift secret that stores the password to login to the PKCS11 session
         LoginSecret string `json:"loginSecret"`
 
-        // +kubebuilder:validation:Optional
-        // The OpenShift secret that stores the HSM certificates.
-        CertificatesSecret string `json:"certificatesSecret,omitempty"`
-
-        // +kubebuilder:validation:Optional
-        // The mounting point where the certificates will be copied to (e.g., /usr/local/luna/config/certs).
-	CertificatesMountPoint string `json:"certificatesMountPoint,omitempty"`
-
-        // +kubebuilder:validation:Optional
-        // +kubebuilder:default=CKM_AES_GCM
-        // Secret encryption mechanism
-	EncryptionMechanism string `json:"encryptionMechanism"`
-
-        // +kubebuilder:validation:Optional
-        // +kubebuilder:default=CKM_AES_KEY_WRAP_KWP
-        // Key wrap mechanism
-	KeyWrapMechanism string `json:"keyWrapMechanism"`
-
-        // +kubebuilder:validation:Optional
-        // +kubebuilder:default=true
-        // Generate IVs for the key wrap mechanism
-	KeyWrapGenerateIV bool `json:"keyWrapGenerateIV"`
-
-        // +kubebuilder:validation:Optional
-        // +kubebuilder:default=true
-        // Generate IVs for CKM_AES_GCM mechanism
-	AESGCMGenerateIV bool `json:"AESGCMGenerateIV"`
-
-        // +kubebuilder:validation:Optional
-        // +kubebuilder:default=true
-        // Always set cka_sensitive
-	AlwaysSetCKASensitive bool `json:"alwaysSetCKASensitive"`
+        // +kubebuilder:validation:Required
+        // The OpenShift secret that stores the HSM client data.
+        // These will be mounted to /var/lib/config-data/hsm
```

Comment:
Where does `/var/lib/config-data/hsm` come from?

Reply:
Per design discussion, we will mount the data in a standard location and use kolla to move the data to the location designated by ClientDataPath.

The path we currently use for kolla stuff is /var/lib/config-data/default.  This is just a branch close to this.

The reason for this is that if we mount the secret to ClientDataPath directly, then we end up overwriting whatever is already there.  So, if we wanted to use /usr/local/luna is the client path, we'd end up deleting everything in that directory and only having the secrets there.